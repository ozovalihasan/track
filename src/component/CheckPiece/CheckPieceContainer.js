import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import CheckPiece from './CheckPiece';
import { fetchCreateTakenTime, fetchDestroyPiece } from '../../redux';

const CheckPieceContainer = ({ piece }) => {
  const dispatch = useDispatch();

  const handleClickCreate = () => {
    dispatch(fetchCreateTakenTime({ piece_id: piece.id }));
  };

  const handleClickDelete = () => {
    dispatch(fetchDestroyPiece(piece.id));
  };

  return (
    <CheckPiece
      piece={piece}
      handleClickCreate={handleClickCreate}
      handleClickDelete={handleClickDelete}
    />
  );
};

CheckPieceContainer.propTypes = {
  piece: PropTypes.shape().isRequired,
};

export default CheckPieceContainer;
