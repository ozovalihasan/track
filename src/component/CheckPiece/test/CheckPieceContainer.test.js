import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import CheckPieceContainer from '../CheckPieceContainer';

const piece = { name: 'Mock Piece Name' };

const mockStore = configureStore();
const store = mockStore();
store.dispatch = jest.fn();

let renderReadyComponent;

beforeEach(() => {
  renderReadyComponent = (
    <Provider store={store}>
      <CheckPieceContainer
        piece={piece}
      />
    </Provider>
  );
});

describe('<CheckPieceContainer />', () => {
  it('triggers handleClickCreate when \'Mock Piece Name\' is clicked', () => {
    render(renderReadyComponent);

    userEvent.click(screen.getByText(/Mock Piece Name/i));
    expect(store.dispatch).toHaveBeenCalled();
  });

  it('triggers handleClickDelete when \'Cancel\' is clicked', () => {
    render(renderReadyComponent);

    userEvent.click(screen.getByText(/Cancel/i));
    expect(store.dispatch).toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const renderedContainer = render(renderReadyComponent);
    expect(renderedContainer).toMatchSnapshot();
  });
});
