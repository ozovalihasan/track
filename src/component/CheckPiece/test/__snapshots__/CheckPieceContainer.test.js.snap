// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckPieceContainer /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ dZUjVD"
      >
        <button
          class="sc-gsTCUz hOQoSq"
          type="button"
        >
          +1
          <div>
            Mock Piece Name
          </div>
        </button>
        <button
          class="sc-dlfnbm qKCXj"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ dZUjVD"
    >
      <button
        class="sc-gsTCUz hOQoSq"
        type="button"
      >
        +1
        <div>
          Mock Piece Name
        </div>
      </button>
      <button
        class="sc-dlfnbm qKCXj"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
